frontend/openchat-agent/src/services/user/mappers.ts

```diff
@@ -1071,8 +1073,8 @@ function proposalAction(action: CandidateProposalAction): ProposalToSubmitAction
         case "transfer_sns_funds":
             return { TransferSnsTreasuryFunds: {
                 to: {
-                    owner: Principal.fromText(action.toPrincipal),
-                    subaccount: []
+                    owner: Principal.fromText(action.recipient.owner),
+                    subaccount: apiOptional((hex) => hexStringToBytes(hex), action.recipient.subaccount)
```

Comment:
We need to pad the string with 0's before converting to bytes, on the BE this needs to be exactly 32 bytes.

Reply:
Or add the 0's after converting to bytes...

Reply:
Yes of course